src/Nat.mo

```diff
@@ -1,72 +1,299 @@
-/// Natural numbers
+/// Utility functions for working with natural numbers.
 ///
 /// Most operations on natural numbers (e.g. addition) are available as built-in operators (e.g. `1 + 1`).
 /// This module provides equivalent functions and `Text` conversion.
+///
+/// Import from the base library to use this module.
+/// ```motoko name=import
+/// import Nat "mo:base/Nat";
+/// ```
 
 import Int "Int";
 import Order "Order";
 import Prim "mo:⛔";
+import Char "Char";
 
 module {
 
   /// Infinite precision natural numbers.
   public type Nat = Prim.Types.Nat;
 
-  /// Conversion.
-  public let toText : Nat -> Text = Int.toText;
+  /// Converts a natural number to its textual representation.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// Nat.toText 1234 // => "1234"
+  /// ```
+  public func toText(n : Nat) : Text = Int.toText n;
+
+  /// Creates a natural number from its textual representation. Returns `null`
+  /// if the input is not a valid natural number.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// Nat.fromText "1234" // => ?1234
+  /// ```
+  public func fromText(text : Text) : ?Nat {
```

Comment:
Just noticed that we do not have this function in `Int`. Maybe it would be useful to offer it there two (and even combine the implementation, similar to `toText`).

Reply:
#520 Yeah that sounds good!